File: my-folder/0004-median-of-two-sorted-arrays/solution.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int sum = nums1.length + nums2.length;
        int[] res = new int[sum];
        int i = nums1.length - 1;
        int j = nums2.length - 1;
        int k = nums1.length + nums2.length - 1;
        int len = k;
        while (j>=0){
            if( i >= 0 && nums1[i] > nums2[j]){
                res[k] = nums1[i];
                i--;
                k--;
            }
            else{
                res[k] = nums2[j];
                j--;
                k--;
            }      
        }
        while(i >= 0){
            res[k] = nums1[i];
            i--;
            k--;
        }
        if(len % 2 == 1){
            return (double)(res[len/2] + res[(len/2)+1])/2;
        }else{
            return (double)res[len/2];
        }
    }
}

File: my-folder/0004-median-of-two-sorted-arrays/solution.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        combinedLength = len(nums1) + len(nums2)
        index = combinedLength // 2
        count = i = j = 0
        prev = curr = None

        while count <= index:
            if i < len(nums1) and (j >= len(nums2) or nums1[i] <= nums2[j]):
                prev = curr
                curr = nums1[i]
                i += 1
            else:              
                prev = curr
                curr = nums2[j]
                j += 1
            count += 1
        if combinedLength % 2 == 0:
            return (prev + curr)/2
        else:
            return curr
